### PR TITLE
Make abacard great again

### PR DESCRIPTION
**Pull request by LudvigHz**
# Description

Fix abacard

# Result

It now works as expected.

# Testing

- [x] I have thoroughly tested my changes.

I tested both /validator and abacard on event pages.

---

Resolves ABA-305

### CODE REVIEW

File: app/components/Search/SearchPageResults.css [hunk 1]
```diff
@@ -37,6 +37,8 @@
 
 .searchResultItemIcon {
   margin-left: 10px;
+  font-size: 0.3rem;
```
Comment (ivarnakken) [inline app/components/Search/SearchPageResults.css:40]: The icons comp has a `size` prop u know

File: app/components/UserValidator/index.tsx [hunk 2]
```diff
@@ -15,16 +15,28 @@ import type { Required } from 'utility-types';
...
+  payload: unknown;
+};
+
+type Props = Omit<
+  ComponentProps<typeof SearchPage<UserSearchResult>>,
+  'handleSelect'
+> & {
   clearSearch: () => void;
-  handleSelect: (arg0: UserWithUsername) => Promise<User>;
+  handleSelect: (arg0: UserWithUsername) => Promise<User | Res>;
```
Comment (ivarnakken) [inline app/components/UserValidator/index.tsx:27]: Couldn't name this something that made more sense? Like `unknownResponse` or something

File: app/components/UserValidator/index.tsx [hunk 3]
```diff
@@ -15,16 +15,28 @@ import type { Required } from 'utility-types';
...
   onQueryChanged: (arg0: string) => void;
   results: Array<UserSearchResult>;
   searching: boolean;
-} & ComponentProps<typeof SearchPage<UserSearchResult>>;
+  validateAbakusGroup: boolean;
+};
+
+const isUser = (user: User | Res): user is User => {
+  return 'username' in user;
+};
```
Comment (ivarnakken) [inline app/components/UserValidator/index.tsx:36]: 💯

File: app/routes/events/EventAbacardRoute.ts [hunk 4]
```diff
@@ -23,7 +23,7 @@ const loadData = async (props, dispatch): any => {
 };
 
 const mapStateToProps = (state, props) => {
-  const query = qs.parse(props.location.search);
+  const query = qs.parse(props.location.search).q;
```
Comment (ivarnakken) [inline app/routes/events/EventAbacardRoute.ts:26]: what this do

File: app/routes/events/components/EventAdministrate/Abacard.tsx [hunk 5]
```diff
@@ -26,7 +27,11 @@ const Abacard = (props: Props) => {
     (reg) => reg.presence === 'PRESENT' && reg.pool
   ).length;
 
-  const handleSelect = ({ username = '' }: { username?: string }) =>
+  const handleSelect = ({
+    username,
+  }: {
+    username: string;
+  }): Promise<{ payload: unknown }> =>
```
Comment (ivarnakken) [inline app/routes/events/components/EventAdministrate/Abacard.tsx:34]: use that `res` thingy

**Review by ivarnakken — APPROVED** [review]
im hella tired but this looks solid

:shipit:

File: app/components/Search/SearchPageResults.css (hunk 1 above)
Comment (LudvigHz) [reply app/components/Search/SearchPageResults.css:40]: This is for the Alt text. Not the image. It needs to be really small to not look crazy as hell

File: app/components/UserValidator/index.tsx (hunk 2 above)
Comment (LudvigHz) [reply app/components/UserValidator/index.tsx:27]: Yes

File: app/routes/events/EventAbacardRoute.ts (hunk 4 above)
Comment (LudvigHz) [reply app/routes/events/EventAbacardRoute.ts:26]: It's the quary parameter for the search, eg. /abacard?q=Ludvig

This is an unrelated bug also fixed here with the search results not clearling when selecting a user

**Pull request merged**